client/lobbies/view.jsx

```diff
@@ -129,6 +134,7 @@ export default class LobbyView extends React.Component {
           lobby={lobby.info}
           chat={lobby.chat}
           user={user}
+          isFavoritingMap={maps.favoriteStatusRequests.has(lobby.info.map.id)}
```

Comment:
I don't think the `favoriteStatusRequests` thing is new but probably a better way to handle this is to update the map's state to favorited, then set it back to unfavorited if that request fails (and vice versa). Makes the handling a lot simpler here and everywhere else that has a favorite icon

Reply:
The only reason why I added this is to disable the favorite/unfavorite button while the request hasn't finished yet. Not sure what would be the best way to deal with the user pressing a button again while the first request hasn't finished yet with that approach?

I guess it would still be worth doing since it is pretty annoying to use this state every time you want to add a favorite icon on map.

Reply:
I don't think you really *need* to disable that. If you submit two requests in a row they're pretty likely to be delivered in that order and succeed in that order. The main complexity is what happens if many fail in succession, but I think you can solve that by keeping track of the "last known state" which is either what you last retrieved from the server directly, or what you know succeeded. If you retrieve from the server you update that value, if a request promise succeeds you update that value. If the last outstanding promise fails (and *only* if it's the last outstanding one), you reset the current value to whatever that last known state is.

imo this current state looks very much like internal state that things generally shouldn't be accessing, which is why I'm wary of needing to pass it to UI elements, especially frequently.